spreadsheet/sheet.go

```diff
@@ -29,6 +29,12 @@ type Sheet struct {
 	w   *Workbook
 	cts *sml.CT_Sheet
 	x   *sml.Worksheet
+	maxColumnIdx uint32
+}
+
+// MaxColumnIdx returns the max used column of the sheet.
+func (s Sheet) MaxColumnIdx() uint32 {
+	return s.maxColumnIdx
```

Comment:
Wouldn't it be easier to find the max column in this function rather than keeping it up to date on every change?
Easy to miss if there is a change and it doesn't get called.
This function should be only called once so should be a pretty cheap operation to search for it here.

Reply:
@gunnsth in order to find the max column we need to make iterations which you asked earlier to get rid of as being wasteful, it was in 0238d3ef , return it?

Reply:
Not so wasteful if we only do it once.  Only need to call this once to get the max column prior
to looping through all rows.

Reply:
fixed